micrometer-core/src/main/java/io/micronaut/configuration/metrics/util/MetricOptionsUtil.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.metrics.util;
+
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.configuration.metrics.annotation.MetricOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class for {@link MetricOptions}. Sharing means of evaluating condition
+ *
+ * @since 5.10.0
+ * @author Haiden Rothwell
+ */
+public class MetricOptionsUtil {
```

Comment:
can be `final` and probably doesn't need to be `public`

Reply:
when not `public` it is not callable from the interceptors due to differing packages